src/internal/deployment.js

```diff
@@ -39,11 +41,22 @@ class Deployment {
     this.githubServerUrl = context.githubServerUrl
     this.artifactName = context.artifactName
     this.isPreview = context.isPreview === true
+    this.timeout = maxTimeout
+    this.startTime = null
   }
 
   // Ask the runtime for the unsigned artifact URL and deploy to GitHub Pages
   // by creating a deployment with that artifact
   async create(idToken) {
+    if (Number(core.getInput('timeout')) > maxTimeout) {
+      core.warning(
+        `Warning: timeout value is greater than the allowed maximum - timeout set to the maximum of ${maxTimeout} milliseconds.`
+      )
+    }
+
+    let timeoutInput = Number(core.getInput('timeout'))
```

Comment:
🧹 **Cleanup**

Not a big deal but we may as well just parse the input once. 🤷🏻 

Also, we can use a `const` keyword instead of a `let` since we aren't changed that value later.

```suggestion
    const timeoutInput = Number(core.getInput('timeout'))

    if (timeoutInput > maxTimeout) {
      core.warning(
        `Warning: timeout value is greater than the allowed maximum - timeout set to the maximum of ${maxTimeout} milliseconds.`
      )
    }

```